src/Infrastructure/Manifest/Interaction/ManifestInteractionProcessor.php

```diff
@@ -0,0 +1,200 @@
+
+    /**
+     * @var \SprykerSdk\Sdk\Infrastructure\Manifest\Interaction\Question\NeedToAskQuestion
+     */
+    protected NeedToAskQuestion $needToAskQuestion;
+
+    /**
+     * @var \SprykerSdk\Sdk\Infrastructure\Manifest\Interaction\Question\ValueQuestion
+     */
+    protected ValueQuestion $valueQuestion;
+
+    /**
+     * @var \SprykerSdk\Sdk\Infrastructure\Manifest\Interaction\Question\NewCollectionItemQuestion
+     */
+    protected NewCollectionItemQuestion $newCollectionItemQuestion;
+
+    /**
+     * @param \SprykerSdk\Sdk\Infrastructure\Manifest\Interaction\Question\NeedToAskQuestion $needToAskQuestion
+     * @param \SprykerSdk\Sdk\Infrastructure\Manifest\Interaction\Question\ValueQuestion $valueQuestion
+     * @param \SprykerSdk\Sdk\Infrastructure\Manifest\Interaction\Question\NewCollectionItemQuestion $newCollectionItemQuestion
+     */
+    public function __construct(
+        NeedToAskQuestion $needToAskQuestion,
+        ValueQuestion $valueQuestion,
+        NewCollectionItemQuestion $newCollectionItemQuestion
+    ) {
+        $this->needToAskQuestion = $needToAskQuestion;
+        $this->valueQuestion = $valueQuestion;
+        $this->newCollectionItemQuestion = $newCollectionItemQuestion;
+    }
+
+    /**
+     * @param array $valueConfigMap
+     *
+     * @return array
+     */
+    public function receiveValues(array $valueConfigMap): array
+    {
+        return $this->receiveValue('', $valueConfigMap, new SplStack(), []);
+    }
+
+    /**
+     * @param string $id
+     * @param \SprykerSdk\Sdk\Infrastructure\Manifest\Interaction\Config\InteractionValueConfig|array $valueConfig
+     * @param \SplStack $keyStack
+     * @param array $receivedValues
+     *
+     * @throws \InvalidArgumentException
+     *
+     * @return array
+     */
+    protected function receiveValue(string $id, $valueConfig, SplStack $keyStack, array $receivedValues): array
+    {
+        switch (true) {
+            case $valueConfig instanceof ReceivedValue:
+                $value = $this->processReceivedValue($id, $valueConfig);
+
+                return $value !== static::NOT_SET_VALUE
+                    ? $this->setValueByKeyStack($receivedValues, $keyStack, $value)
+                    : $receivedValues;
+            case $valueConfig instanceof StaticValue:
+                return $this->setValueByKeyStack($receivedValues, $keyStack, $valueConfig->getValue());
+            case $valueConfig instanceof ValueCollection:
+                return $this->processCollection($id, $valueConfig, $keyStack, $receivedValues);
+            case $valueConfig instanceof CallbackValue:
+                return $this->processCallbackValue($id, $valueConfig, $keyStack, $receivedValues);
+            case is_array($valueConfig):
+                return $this->processNested($valueConfig, $keyStack, $receivedValues);
+            default:
+                throw new InvalidArgumentException(sprintf('Invalid mapping in `%s` node', $id));
+        }
```

Comment:
We don't need polymorphism there due to small class and recursion

Reply:
Unfortunately, I must insist on polymorphic approach